### PR TITLE
feat: step animation polishing

### PR DESCRIPTION
**Pull request by guerrap**

### CODE REVIEW

File: packages/next-frontend/src/components/step/index.tsx [hunk 1]
```diff
@@ -18,9 +18,14 @@ export function Step({
     disabled,
     children,
 }: StepProps) {
+    const wrapperRef = useRef<HTMLDivElement>(null);
     const [open, setOpen] = useState(false);
-    const [height, setHeight] = useState<Height>(0);
-    const [heightAnimationDone, setHeightAnimationDone] = useState(true);
+    const [spring, animate] = useSpring(
+        () => ({
+            height: "83px",
```
Comment (luzzif) [inline packages/next-frontend/src/components/step/index.tsx:25]: Could we maybe use REMs here?

File: packages/next-frontend/src/components/step/index.tsx [hunk 2]
```diff
@@ -31,44 +36,37 @@ export function Step({
         matchChildByType(child, StepContent),
     ) as ReactElement;
 
+    useEffect(() => {
+        animate({
+            height: (open ? wrapperRef?.current?.offsetHeight : 83) + "px",
```
Comment (luzzif) [inline packages/next-frontend/src/components/step/index.tsx:41]: No way to use auto here? Not particularly loving the explicit pixels...

File: packages/next-frontend/src/components/step/index.tsx [hunk 3]
```diff
@@ -31,44 +36,37 @@ export function Step({
...
-            </div>
-            <AnimateHeight
-                height={height}
-                duration={200}
-                easing="ease-in-out"
-                onHeightAnimationEnd={handleContentAnimationOnEnd}
-            >
+            <div ref={wrapperRef}>
+                <div onClick={handlePreviewOnClick}>
+                    {React.cloneElement<StepPreviewProps>(previewChildren, {
```
Comment (luzzif) [inline packages/next-frontend/src/components/step/index.tsx:64]: Why do we need the clone here? To pass a prop?

**Review by luzzif — CHANGES_REQUESTED** [review]

File: packages/next-frontend/src/components/step/index.tsx (hunk 1 above)
Comment (guerrap) [reply packages/next-frontend/src/components/step/index.tsx:25]: Using rems here causes strange behavior on the height.

File: packages/next-frontend/src/components/step/index.tsx (hunk 2 above)
Comment (guerrap) [reply packages/next-frontend/src/components/step/index.tsx:41]: There's an issue while trying to animate between number and `auto` here:
>You are trying to animate between an AnimatedValue and an AnimatedString. In React Spring.

File: packages/next-frontend/src/components/step/index.tsx (hunk 3 above)
Comment (guerrap) [reply packages/next-frontend/src/components/step/index.tsx:64]: Yes, here we need to pass `open` to the `preview` child component, to apply the proper styles.

Comment (luzzif) [reply packages/next-frontend/src/components/step/index.tsx:64]: And isn't it possible to do this without cloning by properly typing the `previewChildren` as a function coomponent that takes in that prop?

**Pull request merged**